### PR TITLE
[MBL-16141][Student] Share extension | Dashboard notification

### PR DESCRIPTION
**Pull request by kristofnemere**
refs: MBL-16141
affects: Student
release note: none

Test plan: smoke test the student app, the upload notification on dashboard should appear when
- sharing files to the app
- uploading files in the app by the new file upload dialog

The notification has a subtitle when sharing files to the app from outside, then is should be the assignment name

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/main/res/layout/item_dashboard_upload.xml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                android:text="@{itemViewModel.data.subTitle}"
+                android:textColor="@color/textDark"
+                android:textSize="14sp"
+                android:visibility="@{itemViewModel.data.subTitle.empty ? View.GONE : View.VISIBLE}"
+                app:layout_constraintTop_toBottomOf="@id/uploadTitle" />
+
+            <ProgressBar
+                style="@android:style/Widget.ProgressBar.Horizontal"
+                android:layout_width="match_parent"
+                android:layout_height="3dp"
```
Comment (hermannakos) [inline libs/pandautils/src/main/res/layout/item_dashboard_upload.xml:104]: this should be `wrap_content`

File: libs/pandautils/src/main/res/layout/item_dashboard_upload.xml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                android:layout_marginTop="8dp"
+                android:indeterminateOnly="true"
+                app:layout_constraintTop_toBottomOf="@id/uploadSubtitle" />
```
Comment (hermannakos) [inline libs/pandautils/src/main/res/layout/item_dashboard_upload.xml:107]: you can set the progress tint to `@color/backgroundInfo`

**Review by hermannakos — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -99,6 +99,8 @@
     <string name="assignmentSubmissionUpload">Uploading submission for %s</string>
     <string name="assignmentSubmissionError">Submission failed for %s</string>
     <string name="assignmentSubmissionComplete">Successfully submitted %s</string>
+    <string name="dashboardNotificationUploadingSubmissionTitle">Uploading Submission</string>
```
Comment (hermannakos) [inline libs/pandautils/src/main/res/values/strings.xml:102]: These should be moved to pandares.

File: libs/pandautils/src/main/res/layout/item_dashboard_upload.xml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                android:ellipsize="end"
+                android:singleLine="true"
+                android:text="@{itemViewModel.data.subTitle}"
+                android:textColor="@color/textDark"
+                android:textSize="14sp"
+                android:visibility="@{itemViewModel.data.subTitle.empty ? View.GONE : View.VISIBLE}"
+                app:layout_constraintTop_toBottomOf="@id/uploadTitle" />
+
+            <ProgressBar
+                style="@android:style/Widget.ProgressBar.Horizontal"
```
Comment (hermannakos) [inline libs/pandautils/src/main/res/layout/item_dashboard_upload.xml:102]: set it to `style="?android:attr/progressBarStyleHorizontal"`

**Review by hermannakos — CHANGES_REQUESTED** [review]

**Review by hermannakos — APPROVED** [review]

File: libs/pandautils/src/main/res/layout/item_dashboard_upload.xml (hunk 2 above)
Comment (kristofnemere) [reply libs/pandautils/src/main/res/layout/item_dashboard_upload.xml:111]: Fixed, thanks.

File: libs/pandautils/src/main/res/values/strings.xml (hunk 3 above)
Comment (kristofnemere) [reply libs/pandautils/src/main/res/values/strings.xml]: Fixed, thanks.

File: libs/pandautils/src/main/res/layout/item_dashboard_upload.xml (hunk 4 above)
Comment (kristofnemere) [reply libs/pandautils/src/main/res/layout/item_dashboard_upload.xml]: Fixed, thanks.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**